torch/csrc/distributed/autograd/utils.h

```diff
@@ -34,12 +32,21 @@ TORCH_API DistAutogradContext* addRecvRpcBackward(
     std::vector<torch::Tensor>& tensors,
     rpc::worker_id_t fromWorkerId);
 
-// This method is a wrapper utility used internally to attach autograd info
-// and autograd function for each type of rpc call, and then send the message
-std::shared_ptr<torch::distributed::rpc::FutureMessage> sendMessageWithAutograd(
-    torch::distributed::rpc::RpcAgent& agent,
-    const torch::distributed::rpc::WorkerInfo& dst,
-    std::unique_ptr<torch::distributed::rpc::RpcCommandBase> rpcCommand);
+// This method is a wrapper utility used internally to wrap autograd info
+// and attach autograd function for each type of rpc call if it has valid
+// context and tensors require grads, in this case, return RpcWithAutograd
+// message; otherwise return original rpc message.
+TORCH_API rpc::Message getMessageWithAutogradCheck(
+    const rpc::worker_id_t dstId,
+    rpc::Message&& wrappedRpcMsg,
+    rpc::MessageType msgType);
+
+// Send message after autograd checking
+TORCH_API std::shared_ptr<torch::distributed::rpc::FutureMessage> sendMessage(
```

Comment:
nit: call this `sendMessageWithAutograd` since we might attach autograd information here.